src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpSyncNamespaceCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeRefactorings.SyncNamespace;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.SyncNamespace
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.SyncNamespace), Shared]
+    internal sealed class CSharpSyncNamespaceCodeRefactoringProvider :
+        AbstractSyncNamespaceCodeRefactoringProvider<CSharpSyncNamespaceCodeRefactoringProvider, NamespaceDeclarationSyntax, CompilationUnitSyntax>
+    {             
+
+        protected override NamespaceDeclarationSyntax ChangeNamespace(NamespaceDeclarationSyntax node, ImmutableArray<string> namespaceParts)
+        {
+            return node.WithName(CreateNameSyntax(namespaceParts, aliasQualifier: null, namespaceParts.Length - 1).WithTriviaFrom(node.Name));
+        }  
+
+        protected override ImmutableArray<ISymbol> GetDeclaredSymbols(SyntaxNode node, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            SyntaxList<MemberDeclarationSyntax> declarations;
```

Comment:
@mavasani didn't you have a method that give back all decls in a type?  or navigation bar code should have a logic there, can we share that?

Reply:
We have an internal helper [http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Features/CSharpDeclarationComputer.cs,29](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Features/CSharpDeclarationComputer.cs,29) (and also corresponding VB API). However, that computes more then just the declared symbols and can optionally walk the entire tree of declarations.

---
In reply to: [221012374](https://github.com/dotnet/roslyn/pull/30156#discussion_r221012374) [](ancestors = 221012374)